src/utils/PSMFeatureExtractor.cpp

```diff
@@ -114,7 +114,9 @@ class PSMFeatureExtractor :
     registerInputFileList_("in", "<files>", StringList(), "Input file(s)", true);
     setValidFormats_("in", ListUtils::create<String>("mzid,idXML"));
     registerOutputFile_("out", "<file>", "", "Output file in mzid or idXML format", true);
-    setValidFormats_("out", ListUtils::create<String>("mzid,idXML"));
+    setValidFormats_("out", ListUtils::create<String>("mzid,idXML"));    
+    registerStringOption_("out_type", "<type>", "", "Output file type -- default: determined from file extension or content.", false);
+    setValidStrings_("out_type", ListUtils::create<String>("mzid,idXML"));
```

Comment:
@jpfeuffer is this automatically removed for KNIME nodes or do we need to add that?

Reply:
In KNIME this is automatically added with the setValidFormats above. This appears
in the OutputTypes Tab of the Dialog and adds the appropriate ending to the filename that is generated. Then the tool should be able to infer the correct type. If I am not mistaken, this should have worked for KNIME all the time already. The out_type parameters though does no harm except that you could potentially now give an idXML filename and write mzID.

Reply:
ah ok I thought we filter these params from the config dialog inKNIME